src/gui/statsdialog.cpp

```diff
@@ -90,20 +90,13 @@ void StatsDialog::update()
     // From lt manual: disk_write_queue and disk_read_queue are the number of peers currently waiting on a disk write or disk read
     // to complete before it receives or sends any more data on the socket. It's a metric of how disk bound you are.
 
-    // num_peers is not reliable (adds up peers, which didn't even overcome tcp handshake)
-    const auto torrents = BitTorrent::Session::instance()->torrents();
-    const quint32 peers = std::accumulate(torrents.cbegin(), torrents.cend(), 0, [](const quint32 acc, const BitTorrent::TorrentHandle *torrent)
-    {
-        return (acc + torrent->peersCount());
-    });
-
     m_ui->labelWriteStarve->setText(QString("%1%")
-                                    .arg(((ss.diskWriteQueue > 0) && (peers > 0))
-                                         ? Utils::String::fromDouble((100. * ss.diskWriteQueue) / peers, 2)
+                                    .arg(((ss.diskWriteQueue > 0) && (ss.peersCount > 0))
```

Comment:
Lets also clean up and simplify the statement a bit:
```c++
m_ui->labelWriteStarve->setText((ss.peersCount > 0)
    ? Utils::String::fromDouble((100. * ss.diskWriteQueue / ss.peersCount), 2)
    : QLatin1String("0"));  // or QString if QLatin1String didn't work
```

Reply:
>Lets also clean up and simplify the statement a bit:

However this might produce "0.00" when `ss.diskWriteQueue == 0` which is different than before. So we need to add `if (n == 0) return QLatin1String("0");` at the top in `Utils::String::fromDouble()` function.
This is a little bit more work than expected, so it is up to you.

Reply:
The `(ss.diskWriteQueue > 0) && (peers > 0)` that you want to change to just `(peers > 0)` made it so that the result was `0` even if fromDouble would produce `0.00`. I think we should not change `fromDouble` - it is less surprising if an input of "zero" also produces an output that respects the precision argument.

Reply:
I have split the statement into multiple lines but I have not applied the change to `fromDouble`. Let me know what you decide. I should note that respecting the precision even when the input is zero matches the behaviour of `QString::number()`.

```cpp
QString::number(0, 'f', 1); // 0.0
QString::number(0, 'f', 2); // 0.00
QString::number(0.0, 'f', 1); // 0.0
QString::number(0.0, 'f', 2); // 0.00
```

Reply:
>it is less surprising if an input of "zero" also produces an output that respects the precision argument.

You are right on this... However I find it a bit weird that we would expect seeing "0" instead of "0.00" when all other values are presented in decimal number.

Reply:
IMO, the way it is now, we pay this small price for the following gains:

- "zero" is displayed in the GUI as `0`, which is consistent with the display of all the other stats (with the obvious exception of global ratio, of course)
- `Utils::fromDouble()` remains consistent with `QString::number()` in respecting the precision argument